### PR TITLE
Tag support for lambda.

### PR DESCRIPTION
**Pull request by bazimov**
AWS recently added tagging support for lambda. This adds tags for lambda deployments.

### CODE REVIEW

File: src/foremast/args.py [hunk 1]
```diff
@@ -44,10 +44,27 @@ def add_env(parser):
...
+    """Add a `group` flag to the _parser_."""
+    parser.add_argument('-g',
+                        '--group',
+                        choices=ENVS,
+                        default=os.getenv('PROJECT', 'group'),
+                        help='Git project name')
+
+
+def add_repo(parser):
+    """Add a `group` flag to the _parser_."""
```
Comment (ckamradt) [inline src/foremast/args.py]: copypasta

**Comment by ndcampbell** [comment]
@bazimov Most of this is not actually needed. You do not need to pass in repo or group. 

You have a generator object at https://github.com/gogoair/foremast/blob/8039094bffd9c050a2c18b95272dabd1014cefcc/src/foremast/awslambda/awslambda.py#L47 and you can extract group and app name from that.

You can see that code at https://github.com/gogoair/gogo-utils/blob/master/src/gogoutils/generator.py

You shouldnt need to touch any other files except `awslambda.py` to add this feature.

File: src/foremast/awslambda/__main__.py [hunk 2]
```diff
@@ -21,7 +21,7 @@
 import argparse
 import logging
 
-from ..args import add_app, add_debug, add_env, add_properties, add_region
+from ..args import add_app, add_debug, add_env, add_properties, add_region, add_group, add_repo
```
Comment (ndcampbell) [inline src/foremast/awslambda/__main__.py]: As per my comment, neither of these are needed.

File: src/foremast/awslambda/awslambda.py [hunk 3]
```diff
@@ -40,6 +40,8 @@ def __init__(self, app, env, region, prop_path):
         """
         self.app_name = app
         self.env = env
+        self.group = group
```
Comment (ndcampbell) [inline src/foremast/awslambda/awslambda.py]: Not needed, you can get this info from `generated`

File: src/foremast/awslambda/awslambda.py [hunk 4]
```diff
@@ -207,7 +209,6 @@ def create_function(self, vpc_config):
         with zipfile.ZipFile(zip_file, mode='w') as z:
             z.writestr('index.py', 'print "Hello world"')
 
-        contents = ''
```
Comment (ndcampbell) [inline src/foremast/awslambda/awslambda.py]: Why did you remove this line? Are you sure their are no sideeffects from this?

File: src/foremast/awslambda/awslambda.py [hunk 5]
```diff
@@ -227,7 +228,13 @@ def create_function(self, vpc_config):
                 Timeout=int(self.timeout),
                 MemorySize=int(self.memory),
                 Publish=False,
-                VpcConfig=vpc_config)
+                VpcConfig=vpc_config,
+                Tags={
+                    'app_group': self.group,
+                    'app_name': self.repo,
```
Comment (ndcampbell) [inline src/foremast/awslambda/awslambda.py]: It looks like that we (Gogo) use the full app_name for the `app_name` tag, not just the repo name. Lets keep this consistent with EC2 tags.

File: src/foremast/runner.py [hunk 6]
```diff
@@ -175,7 +175,9 @@ def create_awslambda(self):
         awslambdaobj = awslambda.LambdaFunction(app=self.app,
                                                 env=self.env,
                                                 region=self.region,
-                                                prop_path=self.json_path)
+                                                prop_path=self.json_path,
+                                                group=self.group,
```
Comment (ndcampbell) [inline src/foremast/runner.py]: group and repo not needed

**Comment by bazimov** [comment]
Thank you @ndcampbell! It was much simpler than I thought. Rebased my last commit with suggestions you have given.

**Comment by ndcampbell** [comment]
lgtm :)
